bokeh/io.py

```diff
@@ -413,6 +414,8 @@ def _get_save_args(state, filename, resources, title):
     if filename is None:
         warn = False
         filename = _detect_filename("html")
+        if not os.access(filename, os.W_OK):
```

Comment:
@jsignell the syntax is a bit confusing, `W_OK | X_OK` is bitwise-or'ing two flags together. It's actually stating the directory has to be both writeable *and* "executable" (i.e. listable, for a dir) .